app/javascript/article-form/components/Options.jsx

```diff
@@ -65,6 +91,36 @@ export const Options = ({
       </div>
     );
   }
+  if (schedulingEnabled) {
+    publishedAtField = (
+      <div className="crayons-field mb-6">
+        <label htmlFor="publishedAt" className="crayons-field__label">
+          Schedule Publication
+        </label>
+        <input
+          type="datetime-local"
+          min={minPublishedAt}
+          value={localPublishedAt} // "2022-04-28T15:00:00"
+          readonly={readonlyPublishedAt}
+          className="crayons-textfield"
+          name="publishedAt"
+          onChange={onConfigChange}
```

Comment:
I ended up using `onChange` here, which means that partial input is not allowed, so if the user only puts date, or date + hour + minutes, the input won't be saved.
We had a discussion with @aitchiss about it, but at that point I thought that the partial input is at least passed to the backend, but I was wrong about it (we discussed changing the button title "Publish"/"Schedule" when the current or future date is selected).
The current solution may be unclear to the user in case of partial input. We could add a tip for the user at least to be more clear, or use smth different instead of `datetime-local` input (e.g. multiple selects, but I hate this idea :D) @aitchiss  what do you think?

Reply:
Ugh I hate these datetime inputs 🙈 

As far as I can see there is no way for us to distinguish between:
- An empty date input because the user doesn't care about scheduling time
- A partially completed date input because the user didn't know they have to be specific to the minute

I thought we could maybe "back-fill" the time, but even if we try to access the input value in JS, a partial input just comes back as an empty string 😞). I suspect this is why we have gone for the multiple selects route in various places in the admin area (e.g. `/admin/advanced/sponsorships/:id/edit`.

The only thing we can do, like you say, is add some description under the input that says you have to fill out everything including the time in order for it to save. Maybe we could even have some dynamic text under that to say e.g. "Your post is set to publish immediately / Your post is scheduled to be published at XXX", just to try to be as explicit as possible.

I think whatever route we choose to go down is probably a product decision, since it's not as straightforward as initially thought. My two cents is that I don't think it's good UX to use this datetime input when we can't offer good error handling for a partially filled input 🤷‍♀️ But maybe with some explicit messaging it could be "good enough" until we iterate further, IDK.

In terms of an alternative to the datetime input, I think we have two options:

**1. Multiple selects**

Like you say, not a beautiful UI at all, and I think possibly some horrible validation required (e.g. those date pickers in admin area let you select 31 February 🙃)

**2. Create a Single Date Picker based on `react-dates`**

We have a (very new) [DateRangePicker](https://storybook.forem.com/?path=/story/beta-daterangepicker--default) based on `react-dates`, and we could use the library to create a single date picker too (e.g. [here's their 'default'](https://react-dates.github.io/react-dates/?path=/story/singledatepicker-sdp--default)). It looks like we'd need to customise it a bit to support the time picking.

The downside to this is that it might take a while since it's creating a whole new design system component.

Reply:
I think, ideally, we should go with the react datetime picker. But at this point this may postpone the feature release :thinking: , @jeremyf what do you think?

Reply:
We can pre-populate, but only the full value with date and time. What I don't like about the solution it's that the `published_at` will be set to the time when the user opens the editor (if the user doesn't modity it), so by the time the user publishes, the `published_at` will be in the past. Currently, if the user doesn't specify the value, the `published_at` is set to the time when the user publishes, which is the desired behavior.

Reply:
Yeah when I was thinking this through I also discounted pre-filling it for the same reasons you mention @lightalloy. I think also it could be alarming if you open this pop-up to change some other setting and then see a form field that suggests your post is going live _immediately_ 🚨  when what you actually want is for that field to be reassuringly blank.